transport.go

```diff
@@ -209,17 +214,31 @@ func (wv webhookValidator) setWebhookDefaults(webhook *Webhook, requestOriginHos
 
 }
 
-func errorEncoder(_ context.Context, err error, w http.ResponseWriter) {
-	w.Header().Set(contentTypeHeader, jsonContentType)
-	code := http.StatusInternalServerError
-	var sc kithttp.StatusCoder
-	if errors.As(err, &sc) {
-		code = sc.StatusCode()
+func errorEncoder(getLogger GetLoggerFunc) kithttp.ErrorEncoder {
+	if getLogger == nil {
+		getLogger = func(_ context.Context) log.Logger {
+			return nil
+		}
 	}
-	w.WriteHeader(code)
 
-	json.NewEncoder(w).Encode(
-		map[string]interface{}{
-			"message": err.Error(),
-		})
+	return func(ctx context.Context, err error, w http.ResponseWriter) {
+		w.Header().Set(contentTypeHeader, jsonContentType)
+		code := http.StatusInternalServerError
+		var sc kithttp.StatusCoder
+		if errors.As(err, &sc) {
+			code = sc.StatusCode()
+		}
+
+		logger := getLogger(ctx)
+		if logger != nil && code != http.StatusNotFound {
+			logger.Log("sending non-200, non-404 response", err, code)
```

Comment:
```suggestion
			logger.Log("sending non-200, non-404 response", level.Key(), level.ErrorValue(), err, code)
```
We need to provide a level for the log.
(level is `"github.com/go-kit/kit/log/level"`)